### PR TITLE
Removed copy_feature_matrix method from DenseFeatures

### PR DESCRIPTION
**Pull request by lambday**
@karlnapf turns out, clone was required (one unit-test failed otherwise). So, put the cloning responsibility in the caller.

### CODE REVIEW

File: src/shogun/clustering/KMeans.cpp [hunk 1]
```diff
@@ -71,7 +71,8 @@ void CKMeans::Lloyd_KMeans(SGMatrix<float64_t> centers, int32_t num_centers)
 				   	Terminating. \n", iter)
 
 		changed=0;
-		rhs_mus->copy_feature_matrix(centers);
+		rhs_mus->set_feature_matrix(centers.clone());
```
Comment (karlnapf) [inline src/shogun/clustering/KMeans.cpp]: so it does need to be cloned in the end?

File: src/shogun/clustering/KMeans.cpp [hunk 2]
```diff
@@ -71,7 +71,8 @@ void CKMeans::Lloyd_KMeans(SGMatrix<float64_t> centers, int32_t num_centers)
...
+		rhs_mus->initialize_cache();
```
Comment (karlnapf) [inline src/shogun/clustering/KMeans.cpp]: this is a new call?

Comment (karlnapf) [inline src/shogun/clustering/KMeans.cpp]: This shouldnt be called in the setter?

File: src/shogun/clustering/KMeans.cpp (hunk 1 above)
Comment (lambday) [inline src/shogun/clustering/KMeans.cpp]: No clue. Without the clone, one of the unit-test fails. So kept the behavior just the same as before. Maybe Saurabh can fill this in.

File: src/shogun/clustering/KMeans.cpp (hunk 2 above)
Comment (lambday) [inline src/shogun/clustering/KMeans.cpp]: Well, set_feature_matrix was already an existing method. Didn't change its behavior.

**Comment by lambday** [comment]
@karlnapf merging this one then. 

The white-spaces are hard to keep track of. You see, I don't manually do those changes, but my `.vimrc` setup removes all the trailing ones for me whenever it writes the buffer (it is based on the `shogun/doc/wiki/Code-style.md` BTW).

**Comment by karlnapf** [comment]
In the future, pls put whitespace changes to a separate commit. We ask external contributors to do this, so we do it as well. Your vim should not change existing code in my eyes when you save a file.

**Comment by lambday** [comment]
@karlnapf okay.. will use a different vimrc for shogun. not sure whether sending a separate patch for only the white-space changes will ever happen though :P would be much better if we had a hook that does it automatically at the time of merging, and hence preventing them to exist in the first place.. but I don't (yet) know how to do that..

**Comment by karlnapf** [comment]
It is quite easy to put them in a separate commit. When you look at the diff and you realise there are whitespace changes, just undo your actual changes, save, commit, re-do the changes, commit again.
Just 1 min of work for you and makes things so much cleaner.

**Pull request merged**